cmd/geth/dbcmd.go

```diff
@@ -269,44 +322,138 @@ func confirmAndRemoveDB(database string, kind string) {
+}
+
+func hbss2pbss(ctx *cli.Context) error {
+	if ctx.NArg() > 1 {
+		return fmt.Errorf("required arguments: %v", ctx.Command.ArgsUsage)
 	}
-	if ctx.NArg() >= 1 {
-		if d, err := hexutil.Decode(ctx.Args().Get(0)); err != nil {
-			return fmt.Errorf("failed to hex-decode 'prefix': %v", err)
-		} else {
-			prefix = d
+
+	var jobnum uint64
+	var err error
+	if ctx.NArg() == 1 {
+		jobnum, err = strconv.ParseUint(ctx.Args().Get(0), 10, 64)
+		if err != nil {
+			return fmt.Errorf("failed to Parse jobnum, Args[1]: %v, err: %v", ctx.Args().Get(1), err)
 		}
+	} else {
+		// by default
+		jobnum = 1000
 	}
-	if ctx.NArg() >= 2 {
-		if d, err := hexutil.Decode(ctx.Args().Get(1)); err != nil {
-			return fmt.Errorf("failed to hex-decode 'start': %v", err)
-		} else {
-			start = d
+
+	stack, _ := makeConfigNode(ctx)
+	defer stack.Close()
+
+	db := utils.MakeChainDatabase(ctx, stack, false, false)
+	db.Sync()
+	defer db.Close()
+
+	config := trie.HashDefaults
+	triedb := trie.NewDatabase(db, config)
+	triedb.Cap(0)
+	log.Info("hbss2pbss triedb", "scheme", triedb.Scheme())
+	defer triedb.Close()
+
+	headerHash := rawdb.ReadHeadHeaderHash(db)
+	blockNumber := rawdb.ReadHeaderNumber(db, headerHash)
+	if blockNumber == nil {
+		log.Error("read header number failed.")
+		return fmt.Errorf("read header number failed")
+	}
+
+	log.Info("hbss2pbss converting", "HeaderHash: ", headerHash.String(), ", blockNumber: ", *blockNumber)
+
+	var headerBlockHash common.Hash
+	var trieRootHash common.Hash
+
+	if *blockNumber != math.MaxUint64 {
+		headerBlockHash = rawdb.ReadCanonicalHash(db, *blockNumber)
+		if headerBlockHash == (common.Hash{}) {
+			return fmt.Errorf("ReadHeadBlockHash empty hash")
 		}
+		blockHeader := rawdb.ReadHeader(db, headerBlockHash, *blockNumber)
+		trieRootHash = blockHeader.Root
+		fmt.Println("Canonical Hash: ", headerBlockHash.String(), ", TrieRootHash: ", trieRootHash.String())
 	}
+	if (trieRootHash == common.Hash{}) {
+		log.Error("Empty root hash")
+		return fmt.Errorf("Empty root hash.")
+	}
+
+	id := trie.StateTrieID(trieRootHash)
+	theTrie, err := trie.New(id, triedb)
+	if err != nil {
+		log.Error("fail to new trie tree", "err", err, "rootHash", err, trieRootHash.String())
+		return err
+	}
+
+	h2p, err := trie.NewHbss2Pbss(theTrie, triedb, trieRootHash, *blockNumber, jobnum)
+	if err != nil {
+		log.Error("fail to new hash2pbss", "err", err, "rootHash", err, trieRootHash.String())
+		return err
+	}
+	h2p.Run()
+
+	return nil
+}
+
+func pruneHashTrie(ctx *cli.Context) error {
+	if ctx.NArg() != 0 {
+		return fmt.Errorf("required none argument")
+	}
+
 	stack, _ := makeConfigNode(ctx)
 	defer stack.Close()
 
-	db := utils.MakeChainDatabase(ctx, stack, true, false)
+	db := utils.MakeChainDatabase(ctx, stack, false, false)
 	defer db.Close()
 
-	return rawdb.InspectDatabase(db, prefix, start)
+	return rawdb.PruneHashTrieNodeInDataBase(db)
 }
 
-func ancientInspect(ctx *cli.Context) error {
+func inspect(ctx *cli.Context) error {
```

Comment:
why change the function of `inspect`?

from code logic, new logic is totally for different usage

Reply:
Fixed. Made a mistake when rebase code.